### ISSUE TITLE
Reconcile usage of zlib and xml2 in build_visit

### ISSUE THREAD

**Issue by aowen87** [aowen87/BAR#643]
I see we can build our own version of zlib with build_visit. I saw that some other packages in build_visit make use of this if it is specified on the command line, which is good. I also found a dependence on zlib handled with a version from VTK (Xdmf). Similar things are happening with XML2. I found the Xdmf references XML2 from VTK. Uintah also depends on XML2, but doesn't try to use the VTK version and will cause the build
of Uintah to fail.

We should probably use a single version of zlib everywhere and have its use determined by a flag to build_visit, so that it doesn't happen by default. The same applies to XML2, it should be controllable on the command line and used if specified on the command line.

-----------------------REDMINE MIGRATION-----------------------
This ticket was migrated from Redmine. The following information
could not be accurately captured in the new ticket:

Original author: Eric Brugger
Original creation: 09/14/2016 03:04 pm
Original update: 03/16/2017 12:34 pm
Ticket number: 2679
